plugins/keycloak_oidc_login/Module.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\plugins\keycloak_oidc_login;
+
+use app\components\LoginProviderInterface;
+use app\plugins\ModuleBase;
+
+class Module extends ModuleBase
+{
+    public const LOGIN_KEY = 'keycloak_oidc';
+    public const AUTH_KEY_USERS = 'keycloak_oidc';
+
+    private static ?LoginProviderInterface $loginProvider = null;
+
+    public static function getDedicatedLoginProvider(): ?LoginProviderInterface
+    {
+        if (self::$loginProvider === null) {
+            self::$loginProvider = new OidcLogin(
+                'https://keycloak.domain.com',
+                'antragsgruen.domain.com',
+                'supderdupersecret'
```

Comment:
I would love to set these credentials in the config.json - in the Admin panels would be ok too but if this would ever be the only way to log in (is this even possible right now?) then it would difficult to change these values if ever needed.

Reply:
Yea, I don't have a really good way of storing plugin-specific credentials. For the discourse-plugin (✝) I put the configuration in a separate file next to the config.json : https://github.com/CatoTH/antragsgruen/blob/main/plugins/discourse/Module.php#L31

Using the admin panel would work too, using something like https://github.com/CatoTH/antragsgruen/blob/main/plugins/member_petitions/Module.php#L65 , but might be an overkill there